### ISSUE TITLE
Customize @babel/plugin-transform-typescript options

### ISSUE THREAD

**Issue by IlyaSemenov** [unjs/jiti#33]
I would like to be able to pass custom options to `@babel/plugin-transform-typescript`

In particular, for interoperability reasons, I need to support `declare x: string` class members, and for that I need to use this:

```ts
['@babel/transform-typescript', { allowDeclareFields: true }]
```

However, it doesn't seem to be possible to achieve with `opts.babel`. It's not even possible to disable `opts.ts` and re-apply the same plugins manually because you only `push` custom plugins and one of them must be `unshift`ed:

https://github.com/unjs/jiti/blob/f25960af1111401fd1ce2a094bb42e0e868341e9/src/babel.ts#L21-L26

https://github.com/unjs/jiti/blob/f25960af1111401fd1ce2a094bb42e0e868341e9/src/babel.ts#L33-L35

Basically I have to copy/paste the entire `babel.ts` in order to setup typescript options.

### Suggestions

1) `opts.ts` to support not only boolean but also object with `@babel/transform-typescript` options
2) `opts.babel` to support not only subset of options but also a config customization callback which will be called directly before `transformSync`

**Comment by pi0** [comment]
Thanks for describing issue. `allowDeclareFields` is enabled in the latest version since babel 8 will enable it by default too (for nuxt you can recreate lockfile or use `yarn upgrade nuxt`). 

- Supporting plugin options can be a good idea but I need more specific reasons to support them.
- Supporting `babelrc` and callback function per transform is dangerous since we cannot predict cache key anymore

Feel free opening more issues if anything else was missing.

**Comment by IlyaSemenov** [comment]
@pi0 It now crashes with:

```
   ╭────────────────────────────────────────────────────────────────────────────────────────────────╮
   │                                                                                                │
   │   ✖ Nuxt Fatal Error                                                                           │
   │                                                                                                │
   │   Error: UNKNOWN_PLUGIN_PROPERTY: [BABEL] .allowDeclareFields is not a valid Plugin property   │
   │   /Users/semenov/work/eg11/eg11-web/src/frontend/nuxt.config.ts:0:0                            │
   │                                                                                                │
   ╰────────────────────────────────────────────────────────────────────────────────────────────────╯
```

See my comment in https://github.com/unjs/jiti/commit/914499cd58a21ea88eae44f7e1dc830b33e33644

**Comment by pi0** [comment]
Fixed in 0.10.1
